nncf/common/graph/graph.py

```diff
@@ -71,6 +104,49 @@ class NNCFGraphNodeType:
     OUTPUT_NODE = MODEL_OUTPUT_OP_NAME
 
 
+class NNCFGraphEdge:
+    """A structure describing an edge in NNCFGraph. Since nodes of the NNCFGraph are operations
+    on (activation) tensors, an edge in NNCFGraph is a representation of an activation tensor produced or
+    consumed by an operation."""
+    def __init__(self, from_node: NNCFNode, to_node: NNCFNode, tensor_shape: List[int]):
+        """
+        :param from_node: An NNCFNode that sources the directed edge.
+        :param to_node: An NNCFNode that sinks the directed edge.
+        :param tensor_shape: The shape of the activation tensor the edge represents.
+        """
+        self.from_node = from_node
+        self.to_node = to_node
+        self.tensor_shape = tensor_shape
+
+    def __str__(self):
+        return str(self.from_node) + " -> " + str(self.tensor_shape) + " -> " + str(self.to_node)
+
+    def __hash__(self):
+        return hash(str(self))
+
+    def __eq__(self, other):
+        return self.from_node == other.from_node and self.to_node == other.to_node \
+               and self.tensor_shape == other.tensor_shape
+
+
+class NNCFGraphPatternIO:
+    """
+    Describes the inputs and outputs of a subgraph in NNCFGraph.
+    """
+    def __init__(self, input_edges: List[NNCFGraphEdge], output_edges: List[NNCFGraphEdge]):
+        self.input_edges = input_edges
+        self.output_edges = output_edges
+
+
+class NNCFNodeExpression(NodeExpression):
+    """A variation of NodeExpression that has a specific matcher function to work with nxgraph node dicts that
```

Comment:
Please, format the docstring according to the Style Guide.

Reply:
Done